Raven.Client.Lightweight/Listeners/IDocumentSaveChangesListener.cs

```diff
@@ -0,0 +1,20 @@
+//-----------------------------------------------------------------------
+// <copyright file="IDocumentStoreListener.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+using Raven.Json.Linq;
+
+namespace Raven.Client.Listeners
+{
+    /// <summary>
+    /// Hook for users to provide additional logic on store operations
+    /// </summary>
+    /// 
+    public interface IDocumentSaveChangesListener
+    {
+        void BeforeSaveChanges(IDocumentSession session);
+
+        void AfterSaveChanges(IDocumentSession session);
+    }
```

Comment:
I'm not sure that I'm following the point here.
You don't really have any good way to inspect the _state_ of the session, or do something with it.